central/policycategoryedge/datastore/datastore_impl.go

```diff
@@ -0,0 +1,110 @@
+package datastore
```

Comment:
I think we discuss this earlier, but let me draw it out here. There the is a way to avoid datastore layer for edges entirely if you follow the pkg structure below:
```
category
-- datastore
    -- store
         -- category
              -- postgres
         -- edge
              -- postgres

```
Category datastore impl:
```
{
  categoryStorage
  edgeStorage
  ...
}
```
You end up with a much simple structure and less code 😉 . The edge store becomes internal only because it does not need to be exposed as the functionality to set categories is abstracted behind category datastore.

Reply:
Yeah, I don't want to take this on just yet - working for correctness first, optimization later :). But noted :)

Reply:
Since you are not planning on changing the structure RN, you will need to lock edge datastore upserts and deletes.